### PR TITLE
Use `default_rng()` instead of `GLOBAL_RNG` on Julia >= 1.3

### PR DESCRIPTION
**Pull request by devmotion**
I noticed that StatsBase still uses `Random.GLOBAL_RNG` as the default RNG. This PR changes it to `Random.default_rng()` on Julia >= 1.3 (it was introduced in Julia 1.3) which is the new and [documented](https://github.com/JuliaLang/julia/pull/44733) API. On newer Julia versions `Random.GLOBAL_RNG` still exists for backward compatibility but [it has been observed that `Random.GLOBAL_RNG` can lead to reduced performance](https://github.com/JuliaLang/julia/issues/48683).

### CODE REVIEW

File: src/sampling.jl [hunk 1]
```diff
@@ -5,7 +5,13 @@
 #
 ###########################################################
 
-using Random: Sampler, Random.GLOBAL_RNG
+using Random: Sampler
+
+@static if VERSION < v"1.3"
```
Comment (ararslan) [inline src/sampling.jl:10]:
```suggestion
if VERSION < v"1.3.0-DEV.565"
```
`@static` isn't necessary at the top level. Also I tracked down the commit that defined `default_rng` and got its version for maximum ~~pedantry~~ accuracy. Alternatively, we could use `isdefined(Random, :default_rng)` in place of a version check.

Comment (devmotion) [reply src/sampling.jl:10]: Ah, my bad. I read a while ago about unnecessary `@static`s but still sometimes add them a bit too freely 😅 And thanks for looking up the commit, I had decided to just not care about the 1.3 dev versions 😉

File: test/sampling.jl [hunk 2]
```diff
@@ -17,12 +18,17 @@ function test_rng_use(func, non_rng_args...)
...
     @test func(MersenneTwister(1), deepcopy(non_rng_args)...) ==
           func(MersenneTwister(1), deepcopy(non_rng_args)...)
-    # default RNG is Random.GLOBAL_RNG
+    # default RNG is Random.GLOBAL_RNG/Random.default_rng()
     Random.seed!(47)
     x = func(deepcopy(non_rng_args)...)
     Random.seed!(47)
     y = func(Random.GLOBAL_RNG, deepcopy(non_rng_args)...)
     @test x == y
+    if VERSION >= v"1.3"
```
Comment (devmotion) [inline test/sampling.jl:27]:
```suggestion
    if VERSION >= v"1.3.0-DEV.565"
```

File: src/sampling.jl (hunk 1 above)
Comment (ararslan) [reply src/sampling.jl]: AFAIK it doesn't hurt to have it there, it just doesn't have any effect.

> thanks for looking up the commit

Any chance I can get to use Julia's `contrib/commit-name.sh`, I take

**Review by ararslan — APPROVED** [review]

**Pull request merged**